benchmark/cuchembm/datasets/bioactivity.py

```diff
@@ -39,10 +40,43 @@ def __init__(self,
         self.table_name = table_name
         self.properties_cols = properties_cols
         self.max_seq_len = max_seq_len
-        self.raw_data_path = os.path.join(data_dir, 'raw_data.csv')
-        self.filter_data_path = os.path.join(data_dir, 'filtered_data.csv')
+
+        # All directories and files to be used for ExCAPE DB.
+        self.excape_dir = data_dir
+        self.raw_data_path = os.path.join(self.excape_dir, 'publication_inchi_smiles_v2.tsv')
+        self.filter_data_path = os.path.join(self.excape_dir, 'filtered_data.csv')
+        if not os.path.exists(self.excape_dir):
+            os.makedirs(self.excape_dir)
+
+        # Relavent data loaded and computed using input ExCAPE DB.
+        self.smiles = None
+        self.properties = None
+        self.fingerprints = None
+
+    def _download(self):
+        """
+        Download ExCAPE database if not already downloaded.
+        """
+        if not os.path.exists(self.raw_data_path):
+            excape_file = os.path.join(self.excape_dir, 'publication_inchi_smiles_v2.tsv.xz')
+            if not os.path.exists(excape_file):
+                logger.info('Downloading ExCAPE data...')
+                os.system(f'wget -O {excape_file} https://zenodo.org/record/2543724/files/pubchem.chembl.dataset4publication_inchi_smiles_v2.tsv.xz?download=1')
+
+            logger.info(f'Extracting ExCAPE db {excape_file}...')
+            os.system(f'cd {self.excape_dir} && xz -k -d {excape_file}')
```

Comment:
Alternatively can use the python lzma library: https://stackoverflow.com/questions/49348091/iterate-a-large-xz-file-line-by-line-in-python